helix-core/src/line_ending.rs

```diff
@@ -112,6 +107,17 @@ impl LineEnding {
             None
         }
     }
+
+    /// The platform's native line ending.
+    ///
+    /// [`LineEnding::Crlf`] on Windows, otherwise [`LineEnding::LF`].
+    #[inline]
+    pub const fn native() -> Self {
+        #[cfg(target_os = "windows")]
+        return LineEnding::Crlf;
+        #[cfg(not(target_os = "windows"))]
+        return LineEnding::LF;
+    }
```

Comment:
Yup, you're right, it technically does the same thing. I initially removed `DEFAULT_LINE_ENDING` because because it made refactoring much easier - i.e. everything that depends on that constant needs to be changed, since it's not constant anymore.

Apart from that, having it tied to `LineEnding` made more sense to me so I added it there. Though, I don't have a strong preference for this, so if there's a consensus I can change it back to a constant.

Reply:
If it's not absolutely necessary I would prefer if we could keep the constant same as it is currently in master. I think it's readable and used elsewhere and changes like these introduce loads of merge conflicts in other PRs. If there is no functional reason that this needs to be refactored (which I don't think there is) then I would prefer to just keep it as in master an only change the type of the constant.

Reply:
Just changed it back to a `const`.